qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -418,6 +419,42 @@ def name(self) -> str:
         """
         return self._name
 
+    def update_name(self, name: str) -> str:
+        """Update the name of this job set.
+
+        Args:
+            name: The new `name` for this job set.
+
+        Returns:
+            The new name associated with this job set.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the input job name is not a string.
+            IBMQJobManagerUpdateError: If there was an error with updating the names
+                for the jobs in the job set.
+        """
+        if not isinstance(name, str):
+            raise IBMQJobManagerInvalidStateError(
+                '"{}" of type "{}" is not a valid job name. '
+                'The job name needs to be a string.'.format(name, type(str)))
+
+        for i, job in enumerate(self.jobs()):
+            if job:
+                try:
+                    _ = job.update_name("{}_{}_".format(name, i))
+                except IBMQJobUpdateError as ex:
+                    raise IBMQJobManagerUpdateError(
+                        "An error occurred when updating the name for this job set: {}"
+                        "As a result, some of the jobs in this set may not have been "
+                        "updated. Please try updating the job set's name again "
+                        "to update the name for all jobs in this set."
+                        .format(str(ex))) from ex
```

Comment:
Decided to "fail fast" if there is an error with updating the name of a job in the job set, and letting the user know of it. 

Or is it more reasonable to continue updating the name of the rest of the jobs, while logging the exception for the jobs that failed to update. Then, raise an exception in the end, specifying which jobs were not updated?

Note: The case is similar for `update_tags`.

Reply:
Job manager usually issues a warning instead of raising an error if _some_ of the job operations failed. It'd also be nice to denote a job name update failed in the job manager report output (e.g. with an asterisk next to the job name or something)